pkg/bootstrap/platform/azure.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+
+	"istio.io/pkg/log"
+)
+
+const (
+	AzureMetadataEndpoint  = "http://169.254.169.254"
+	AzureInstanceURL       = AzureMetadataEndpoint + "/metadata/instance"
+	AzureDefaultAPIVersion = "2019-08-15"
+	SysVendorPath          = "/sys/class/dmi/id/sys_vendor"
+	MicrosoftIdentifier    = "Microsoft Corporation"
+
+	AzureName     = "azure_name"
+	AzureLocation = "azure_location"
+	AzureVMID     = "azure_vm_id"
+)
+
+var (
+	azureAPIVersionsFn = func() string {
+		return metadataRequest("")
+	}
+	azureMetadataFn = func(version string) string {
+		return metadataRequest(fmt.Sprintf("api-version=%s", version))
+	}
+)
+
+type azureEnv struct {
+	APIVersion      string
+	computeMetadata map[string]interface{}
+	networkMetadata map[string]interface{}
+}
+
+// IsAzure returns whether or not the platform for bootstrapping is Azure
+// Checks the system vendor file (similar to https://github.com/banzaicloud/satellite/blob/master/providers/azure.go)
+func IsAzure() bool {
+	sysVendor, err := ioutil.ReadFile(SysVendorPath)
+	if err != nil {
+		log.Warnf("Error reading sys_vendor in Azure platform detection: %v", err)
```

Comment:
Will this logs a warning when running on other platforms? If so, shall we lower this to debug level? Also, should we just return false here?